rules/TypeDeclaration/Helper/PhpDocNullableTypeHelper.php

```diff
@@ -133,29 +134,38 @@ private function resolveUpdatedPhpDocTypeFromPhpDocTypeAndPhpParserTypeNullInfo(
     ): ?Type {
         /** @var array<(NullType | UnionType)> $updatedDocTypes */
         $updatedDocTypes = [];
-        $phpDocTypeContainsNullType = false;
+        $isPhpDocTypeContainingNullType = false;
+        $isPhpDocTypeContainingClosureType = $phpDocType instanceof ClosureType;
         if ($phpDocType instanceof UnionType) {
-            $phpDocTypeContainsNullType = TypeCombinator::containsNull($phpDocType);
+            $isPhpDocTypeContainingNullType = TypeCombinator::containsNull($phpDocType);
             foreach ($phpDocType->getTypes() as $subType) {
```

Comment:
You can create to separate 2 private methods:

- `hasClosureType()` which loop break on `$subType instanceof ClosureType` return early
- `resolveUpdatedDocType` which loop and append existing `$updatedDocTypes[]` when `$subType` is not `NullType`

so variable can be assigned.